_search-plugins/search-pipelines/rerank-processor.md

```diff
@@ -11,33 +11,49 @@ grand_parent: Search pipelines
-`tag` | String | The processor's identifier. Optional.
-`description` | String | A description of the processor. Optional.
-`ignore_failure` | Boolean | If `true`, OpenSearch [ignores any failure]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/creating-search-pipeline/#ignoring-processor-failures) of this processor and continues to run the remaining processors in the search pipeline. Optional. Default is `false`.
+Field | Data type | Required/Optional | Description
+:--- | :--- | :--- | :---
+`<rerank_type>` | Object | Required | The rerank type for document reranking. Valid values are `ml-opensearch` and `by_field`.
+`context` | Object |  Required for the `ml_opensearch` rerank type. Optional and does not affect the results for the `by_field` rerank type. | Provides the `rerank` processor with information necessary for reranking at query time. 
+`tag` | String | Optional | The processor's identifier.
+`description` | String | Optional | A description of the processor.
+`ignore_failure` | Boolean | Optional | If `true`, OpenSearch [ignores any failure]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/creating-search-pipeline/#ignoring-processor-failures) of this processor and continues to run the remaining processors in the search pipeline. Default is `false`.
+
+<!-- vale off -->
+## The ml_opensearch rerank type
+<!-- vale on -->
+Introduced 2.12
+{: .label .label-purple }
 
-### The `ml_opensearch` reranker type
+To rerank results using a cross-encoder model, specify the `ml_opensearch` rerank type.
 
-The `ml_opensearch` reranker type is designed to work with the cross-encoder model provided by OpenSearch. For this reranker type, specify the following fields.
+### Prerequisite
+
+Before using the `ml_opensearch` rerank type, you must configure a cross-encoder model. For information about using an OpenSearch-provided model, see [Cross-encoder models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#cross-encoder-models). For information about using a custom model, see [Custom local models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/custom-local-models/).
+
+The `ml_opensearch` rerank type supports the following fields. All fields are required.
 
 Field  | Data type | Description
 :--- | :---  | :--- 
-`ml_opensearch` | Object | Provides the rerank processor with model information. Required.
-`ml_opensearch.model_id` | String | The model ID for the cross-encoder model. Required. For more information, see [Using ML models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/using-ml-models/).
-`context.document_fields` | Array | An array of document fields that specifies the fields from which to retrieve context for the cross-encoder model. Required.
+`ml_opensearch.model_id` | String | The model ID for the cross-encoder model to use for reranking. For more information, see [Using ML models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/using-ml-models/).
```

Comment:
```suggestion
`ml_opensearch.model_id` | String | The model ID used by the cross-encoder model for reranking. For more information, see [Using ML models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/using-ml-models/).
```

Reply:
Changed to "of". It's the model ID of the cross-encoder model.